---
output: github_document
---

<!-- README.md is generated from README.Rmd. Please edit that file -->

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  fig.path = "man/figures/README-",
  out.width = "100%"
)
```

# nicheR

<!-- badges: start -->
<!-- badges: end -->

Identify niches of co-occuring cell types from spatial transcriptomics and proteomics data with single cell resolution. `nicheR` implements a convolutional network type approach to identify niches in spatial omics data.

## Installation

You can install the development version of nicheR from [GitHub](https://github.com/MontanaSpiteri/nicheR) with:

``` r
install.packages("devtools")
devtools::install_github("MontanaSpiteri/nicheR")
```

## Usage

```{r usage}
library(nicheR)
```
